retrievalmarket/types/queryaskv2.ipldsch

```diff
@@ -0,0 +1,54 @@
+# Query is a query to a given provider to determine information about a piece
+# they may have available for retrieval
+type Query struct {
+  # If kind is Payload, response will error if not included
+	PayloadCID optional Link
+	# If kind is Piece, response will error if not included
+	PieceCID optional Link 
+}
+
+# QueryResponseStatus indicates whether a queried piece is available
+type QueryResponseStatus enum {
+  # QueryResponseAvailable indicates a provider has a piece and is prepared to
+	# return it
+	| QueryResponseAvailable ("0")
+
+	# QueryResponseUnavailable indicates a provider either does not have or cannot
+	# serve the queried piece to the client
+	| QueryResponseUnavailable ("1")
+
+	# QueryResponseError indicates something went wrong generating a query response
+	| QueryResponseError ("2")
+} representation int
```

Comment:
`string` enums are a lot nicer to see when looking at message contents and they shouldn't cost much more if you choose short names; I think we have a couple of places of regret in the current filecoin stack where int enums exist and we could have gone with strings (IIRC there was a discussion in graphsync when we did the initial bindnode convert). Up to you, not a big deal but the message format here is already fairly verbose and would be pretty nice to look at if you're just decoding the raw messages because of all the maps, but then you get to these numbers and they're not self explanatory.

Reply:
Agree, I'd favour strings 👍